src/VisualStudio/Core/Def/Implementation/ChangeSignature/AddParameterDialogViewModel.cs

```diff
@@ -26,61 +28,124 @@ internal class AddParameterDialogViewModel : AbstractNotifyPropertyChanged
         public AddParameterDialogViewModel(Document document, int insertPosition)
         {
             _notificationService = document.Project.Solution.Workspace.Services.GetService<INotificationService>();
-            _semanticModel = document.GetRequiredSemanticModelAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None);
+            _semanticModel = document.GetRequiredSemanticModelAsync(CancellationToken.None).WaitAndGetResult_CanCallOnBackground(CancellationToken.None);
+
+            TypeIsEmptyImage = Visibility.Visible;
+            TypeBindsImage = Visibility.Collapsed;
+            TypeDoesNotParseImage = Visibility.Collapsed;
+            TypeDoesNotBindImage = Visibility.Collapsed;
+            TypeBindsDynamicStatus = ServicesVSResources.Please_enter_a_type_name;
 
             Document = document;
             InsertPosition = insertPosition;
+
+            IsRequired = true;
+            IsCallsiteRegularValue = true;
+
             ParameterName = string.Empty;
             CallSiteValue = string.Empty;
+            DefaultValue = string.Empty;
         }
 
         public string ParameterName { get; set; }
 
         public string CallSiteValue { get; set; }
 
-        private string TypeNameWithoutErrorIndicator
+        private SymbolDisplayFormat _symbolDisplayFormat = new SymbolDisplayFormat(
+            genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+            miscellaneousOptions: SymbolDisplayMiscellaneousOptions.UseSpecialTypes);
+
+        public ITypeSymbol? TypeSymbol { get; set; }
+
+        public string? TypeName
         {
             get
             {
-                return TypeSymbol!.ToDisplayString();
+                return TypeSymbol?.ToDisplayString(_symbolDisplayFormat);
             }
         }
 
-        private SymbolDisplayFormat _symbolDisplayFormat = new SymbolDisplayFormat(
-            genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
-            miscellaneousOptions: SymbolDisplayMiscellaneousOptions.UseSpecialTypes);
+        public bool TypeBinds => !TypeSymbol!.IsErrorType();
 
-        public ITypeSymbol? TypeSymbol { get; set; }
+        private bool _isRequired;
+        public bool IsRequired
+        {
+            get
+            {
+                return _isRequired;
+            }
```

Comment:
```suggestion
            get => _isRequired;
```

Reply:
There were a couple of these. Fixed them up.